koku/api/query_params.py

```diff
@@ -416,23 +417,36 @@ def _set_tag_keys(self, query_params):
         self.tag_keys = param_tag_keys
 
     def _set_aws_category_keys(self, query_params):
-        """Set the valid aws_category keys"""
+        """Set the valid self.aws_category keys.
+
+        The aws_category_keys variable is used in the report serializer
+        to update the valid field names list. Any key added to this set
+        will not a trigger the unsupport parameter or invalid value error.
+        """
         prefix_list = [AWS_CATEGORY_PREFIX, AND_AWS_CATEGORY_PREFIX, OR_AWS_CATEGORY_PREFIX]
         self.aws_category_keys = set()
```

Comment:
I should have caught this earlier. Avoid creating new attributes in methods. This should be in the constructor so the attribute is always available.